src/components/AnnotationViewer.vue

```diff
@@ -503,6 +534,161 @@ export default class AnnotationViewer extends Vue {
     return newAnnotation;
   }
 
+  private shouldSelectAnnotation(
+    selectionAnnotationType: AnnotationShape,
+    selectionAnnotationCoordinates: any[],
+    annotation: IAnnotation,
+    unitsPerPixel: number
+  ) {
+    const annotationCoordinates = annotation.coordinates;
+
+    if (selectionAnnotationType === AnnotationShape.Point) {
+      // If the selection annotation type is "Point"
+      // Case 1: Annotation to test is a "Point":
+      // The distance point to point should be lower than the annotation radius
+      // Case 2: Annotation to test is a "Line":
+      // The distance between the line and the point should be lower than the lineAnnotation width
+      // Case 3: Annotation to test is "Polygon":
+      // Check if the selection point is positioned into the Polygon.
+
+      const selectionPosition = selectionAnnotationCoordinates[0];
+      const { radius, strokeWidth } = this.getAnnotationStyle(annotation);
+
+      if (annotation.shape === AnnotationShape.Point) {
+        const annotationRadius = (radius + strokeWidth) * unitsPerPixel;
+        const annotationPosition = annotationCoordinates[0];
+        return (
+          pointDistance(selectionPosition, annotationPosition) <
+          annotationRadius
+        );
+      } else if (annotation.shape === AnnotationShape.Line) {
+        // Check if click on points of the line, or on the line directly
+        const width = strokeWidth * unitsPerPixel;
+        return annotationCoordinates.reduce(
+          (isIn: boolean, point: any, index: number) => {
+            let isPointInLine = false;
+            if (index === annotationCoordinates.length - 1) {
+              // Specific case for the last point that does not have a next point
+              isPointInLine = pointDistance(point, selectionPosition) < width;
+            } else {
+              isPointInLine =
+                geojs.util.distance2dToLineSquared(
+                  selectionPosition,
+                  point,
+                  annotationCoordinates[index + 1]
+                ) < width;
+            }
+            return isIn || isPointInLine;
+          },
+          false
+        );
+      } else {
+        return geojs.util.pointInPolygon(
+          selectionPosition,
+          annotationCoordinates
+        );
+      }
+    } else {
+      // If the selection annotation type is "Polygon"
+      // Check if the tested annotation (independently from its type)
+      // is in the defined polygon
+      return annotation.coordinates.reduce((isIn: boolean, point: any) => {
+        const isPointIn = geojs.util.pointInPolygon(
+          point,
+          selectionAnnotationCoordinates
+        );
+        return isIn || isPointIn;
+      }, false);
```

Comment:
it looks like you do not need to browse them all but stop as soon as you found one in. => `annotation.coordinates.some()`

Reply:
We do need to browse them all. 
The idea here, is to get annotations within a manually defined polygon.
We have to go over every annotations and check if it is in the polygon.
We don't want to stop as soon as we found one. We want all of them.

Reply:
If you need ALL to be true (in the polygon), it should have been `isIn && isPointIn` .

In such case, `annotation.coordinates.every()` would be what you are looking for.